### PR TITLE
NFDIV-2553 - Submit AOS citizen bug V2

### PR DESCRIPTION
**Pull request by danielbe11**
### Change description ###

Enter a description.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/main/app/controller/PostController.test.ts [hunk 1]
```diff
@@ -276,6 +279,34 @@ describe('PostController', () => {
 
     expect(res.redirect).toHaveBeenCalledWith('/next-step-url');
   });
+
+  test('getEventName returns UPDATE_AOS when shouldUpdateAos returns true', async () => {
```
Comment (benouaer) [inline src/main/app/controller/PostController.test.ts:283]: I guess the title of the test is not accurate anymore

File: src/main/app/controller/PostController.test.ts [hunk 2]
```diff
@@ -276,6 +279,34 @@ describe('PostController', () => {
...
+    const body = { gender: Gender.MALE };
+    const controller = new PostController(mockFormContent.fields);
+
+    const req = mockRequest({ body });
+    req.session.isApplicant2 = true;
+    req.session.userCase.applicationType = ApplicationType.SOLE_APPLICATION;
+    const res = mockResponse();
+    await controller.post(req, res);
+
+    expect(req.locals.api.triggerEvent).not.toHaveBeenCalledWith('1234', { gender: 'male' }, UPDATE_AOS);
```
Comment (benouaer) [inline src/main/app/controller/PostController.test.ts:308]: is there a test for Draft_AOS?

Comment (danielbe11) [reply src/main/app/controller/PostController.test.ts:308]: https://github.com/hmcts/nfdiv-frontend/pull/2408/files#diff-c1d5768373b57e8a93385dc918e6318a47509092fed497fb43cb6bf71df2b3b4R22

File: src/main/app/controller/PostController.ts [hunk 3]
```diff
@@ -71,7 +72,9 @@ export class PostController<T extends AnyObject> {
   }
 
   protected getEventName(req: AppRequest<T>): string {
-    if (req.session.isApplicant2) {
+    if (shouldUpdateAos(req)) {
```
Comment (benouaer) [inline src/main/app/controller/PostController.ts:75]: we could create a custom UPDATE_AOS post-controller that overrides getEventName, and that is triggered by isAosStep urls
doesn't have to block the P2 though

Comment (danielbe11) [reply src/main/app/controller/PostController.ts:75]: Will make another ticket for this.

**Pull request closed without merging**